_articles/front-proxy.md

```diff
@@ -0,0 +1,227 @@
+will minimize cross-zone traffic (which is expensive and slow), while
+maintaining the ability to incrementally fall back to out-of-zone instances if
+the local ones are failing.
+
+Setting up zone-aware load balancing requires two options to be set:
+
+  - Each cluster (statically defined or returned by the control plane via CDS) must enable zone-aware routing.
+
+  - Each Envoy must set `local_cluster_name` to one of the named clusters.
+
+Statically defined, this looks like:
+
+```yaml
+  local_cluster_name: service1
+  clusters:
+  - name: service1
+    zone_aware_lb_config:
+      min_cluster_size: 3
+```
+
+On the other hand, _fully isolated zones_ should be configured with
+**entirely separate service discovery integrations**. Instead of giving each
+Envoy the full routing table with all instances, it’s better to only enable
+each Envoy to route to instances in its own zone. To enable these zones to fail
+over gracefully, you can add remote front proxies to each cluster with a low
+weight.
+
+This “backhaul” path will allow automatic failover while allowing zones to
+maintain different internal configurations. For example, if you are doing a
+release in the EU region, having the US region fail to the EU’s front proxy
+keeps the US region from inadvertently routing too much traffic to the new
+version if there’s a failure mid-release.
+
+## Best Practices
+
+Once you have a front proxy up and running, there are several ways to make
+operating it easier for you, your team, and the rest of the company.
+
+### Make routing self-serve.
+
+Microservices mean a lot more changes, and service
+teams should be part of the process. Add guardrails (either people process or
+RBAC-style rules) to prevent things from breaking.
+
+### Centralize authentication at the edge.
+
+Adding a single authentication call at  the edge prevents every service from
+re-verifying the user each time. Take the cookie from the user, verify it with
+the authentication service, and store the result (including any commonly used
+data like a user ID) in a header that gets passed to internal services and
+trusted without further verification.
+
+### Add protections from bad actors and unexpected traffic spikes.
+
+This means [retries](automatic-retries.html),
+[health checks](health-check.html), etc. What this looks like depends strongly
+on your infrastructure and the types of issues you’re looking to mitigate.
+
+## Next Steps
+
+While this article has focused on how to handle traffic coming from outside
+your network, it's also possible for Envoy to handle traffic between services
+(or “east-west” traffic). For a lightweight way to set up this internal mesh,
+you can route internal requests through this front proxy (or a similarly
+configured proxy pool specifically for east-west).
+
+Beyond that, you can take better advantage of Envoy’s unique features as a
+lightweight sidecar by setting up a Basic Service Mesh.
```

Comment:
Maybe add a "coming soon!" since we haven't published this article yet?

Reply:
gonna redact and add this back in after service mesh publishes